### PR TITLE
monkey patch np.array for ivy.array conversion 

### PR DESCRIPTION
**Pull request by Aarsh2001**
The following code snippet now works -:

```
import ivy
import numpy as np

for fw in ["jax", "numpy", "torch", "tensorflow"]:
    try:
        ivy.set_backend(fw)
        a = ivy.array(1e-5)
        b = np.array(a)
    except Exception as e:
        print(f"Error in {fw}: {e}")
```

### CODE REVIEW

**Comment by Aarsh2001** [comment]
Looking forward to your thoughts @VedPatwardhan @guillesanbri !🙂

File: ivy/__init__.py [hunk 1]
```diff
@@ -1155,3 +1155,14 @@ def __exit__(self, type, value, traceback):
...
+#We monkey patch np.array to enable ivy.Array conversion with tensorflow backend
+_np_array = np.array
+
+@wraps(_np_array)
+def nparray(*args, **kwargs):
+    if isinstance(args[0], ivy.Array):
+        return ivy.to_numpy(*args, **kwargs)
+    return  _np_array(*args, **kwargs)
+
+np.array = nparray
```
Comment (guillesanbri) [inline ivy/__init__.py:1168]: we shouldn't override the whole `np.array` I guess :sweat_smile:

**Review by guillesanbri — COMMENTED** [review]
Just left a minor comment, overall I think this approach is a little bit nuclear and the overriding shouldn't happen at this level, the idea was more oriented towards finding what internal method is called by `np.array` and then modifying that within the `ivy.Array` class, not sure if that's doable tho

File: ivy/__init__.py (hunk 1 above)
Comment (Aarsh2001) [reply ivy/__init__.py:1170]: Sure will have look. I guess the most granular method is `np.core.multiarray.array(*args, **kwargs)`

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by guillesanbri** [comment]
I think defining a `__array__` method does the trick and allows you to control how `np.array` works for a particular class :slightly_smiling_face:

**Comment by Aarsh2001** [comment]
Thanks @guillesanbri, will have a look and get back to you w results

**Pull request closed without merging**